src/snowflake/snowpark/mock/_plan.py

```diff
@@ -394,7 +394,24 @@ def handle_function_expression(
             current_row
         )  # the row's 0-base index in the window
         to_pass_args.append(row_idx)
-    return _MOCK_FUNCTION_IMPLEMENTATION_MAP[func_name](*to_pass_args)
+
+    result = _MOCK_FUNCTION_IMPLEMENTATION_MAP[func_name](*to_pass_args)
```

Comment:
I'm thinking of a more generalized way to handle this. Sophie has left a comment in the UDF PR and I'm wondering whether the same logic could be applied here: https://github.com/snowflakedb/snowpark-python/pull/1316#discussion_r1532479587.

if we pre-create an empty table/column emulator with size of len(input_data), will the constant data auto fill into the empty dataframe?

Reply:
That approach doesn't work as well here because there is a bit of a difference in how they are applied. 

UDF are applied as a row-wise operation so by initializing a new empty index of size `n` we can insure that `n` calls to the udf happen and `n` values are generated to add to the df.

Built-in functions on the other hand are executed with entire columns at a time which is typically more performant, but also means that without any column inputs a function doesn't know how many values to add.